rfcs/0016-permissionless-assets-for-ibc-and-cw.md

```diff
@@ -0,0 +1,77 @@
+# Overview
+
+To transfer any asset from any chain connected via Centauri to DotSama, this asset needs to be defined in Picasso or Composable assets registry. This operation needs to be permissionless to allow free flow of assets between chains. Moreover, permissionless asset creation needs to support CosmWasm contracts which have functionality of creating, minting and burning tokens. There 2 types of permissionless assets: mintable assets(needed for cosmwasm) and nonmintable(for bridging). This document is an approach how to design permissionless assets and avoid potential risks.
+
+## What we currently have
+
+Currently we support asset creation in AssetsRegistry pallet via Root(Governance). During the permissioned asset creation process, user needs to specify asset's `protocol_id, nonce, name, symbol, decimals, location, ED, ratio`. 
+
+## What we want to achieve
+
+Permissionless assets need to follow the same structure as Permissioned assets and be part of Assets Registry. We want new assets to abide and not break interfaces we have via AssetsRegistry pallet and AssetsTransactorRouter pallet. Permissionless assets will also be modifiable via Root. For permissionless assets' creation a separate pallet is used to become a wrapper for AssetsRegistry: `permissionless_assets`. To create a new asset, user will need to pay registration fee in PICA to register a new asset.
+
+This wrapper will track the creator of the asset and will allow the admin to modify the asset's metadata. The CW assets will also use the admin to allow the account to mint and burn token's assets.
+
+## Potential Risks of Permissionless assets
+
+* **ID manipulation**: Users should not be able to manipulate `protocol_id` and `nonce` values to create ids that conflict with other pallets' ids.
+* **Ratio Manipulation**: Users should not be able to set any ratio value for any asset to be used for BYOG and making XCM transfers with self-set fees. 
+* **ED Manipulation**: Users should not be able to set any Existential Deposits for their assets to avoid bloating the chain's storage
+
+
+## Solutions to the above risks
+
+### ID manipulation
+
+permissionless_assets will have a predefined `protocol_id` and `StorageValue` to increment `nonce` on a new asset's creation to avoid users manipulating new assets' ids and making them conflicted with other pallets' protocol ids.
```

Comment:
protocol id is 32 bit global  network id and 32 bit pallet id.

Reply:
pallet id is 64 bit. we currently dont have global network id in asset id creation. I dont think we need to add it because PalletId + nonce should be enough for pemissionless assets "namespace"